### DISCUSSION TITLE
en/latest/UserGuide/%E8%B7%AF%E7%BD%91%E4%BC%98%E5%8C%96/

### DISCUSSION THREAD

**Discussion by giscus** [zdsjjtTLG/TrackIt#8] [General]
# en/latest/UserGuide/%E8%B7%AF%E7%BD%91%E4%BC%98%E5%8C%96/

在python中轻松进行路径匹配

https://gotrackit.readthedocs.io/en/latest/UserGuide/%E8%B7%AF%E7%BD%91%E4%BC%98%E5%8C%96/

**Comment by Zeitreisender11** [comment]
连通性修复问题：
请问在使用modify_conn函数进行连通性修复，产生以下报错，是因为原始路网的问题么？
有什么好的解决意见，谢谢！
link done
node done
Running connectivity repair...
Traceback (most recent call last):
  File "xxx.py", line 56, in <module>
    link, node = nv.modify_conn(link_gdf=link, node_gdf=node, book_mark_name='topo_fix', generate_mark=True)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xxx/miniconda3/envs/gotrackit_env/lib/python3.11/site-packages/gotrackit/netreverse/NetGen.py", line 437, in modify_conn
    link_gdf, node_gdf = self.fix_minimum_gap(node_gdf=node_gdf, link_gdf=link_gdf)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xxx/miniconda3/envs/gotrackit_env/lib/python3.11/site-packages/gotrackit/netreverse/NetGen.py", line 453, in fix_minimum_gap
    link_gdf, node_gdf, _ = modify_minimum(plain_prj=self.plain_crs, link_gdf=link_gdf, node_gdf=node_gdf,
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xxx/miniconda3/envs/gotrackit_env/lib/python3.11/site-packages/gotrackit/WrapsFunc.py", line 13, in inner
    res = f(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^
  File "/home/xxx/miniconda3/envs/gotrackit_env/lib/python3.11/site-packages/gotrackit/netreverse/RoadNet/SaveStreets/streets.py", line 287, in modify_minimum
    node_degrees_map = {node: node_degrees_dict[node] for node in node_group_list}
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xxx/miniconda3/envs/gotrackit_env/lib/python3.11/site-packages/gotrackit/netreverse/RoadNet/SaveStreets/streets.py", line 287, in <dictcomp>
    node_degrees_map = {node: node_degrees_dict[node] for node in node_group_list}
                              ~~~~~~~~~~~~~~~~~^^^^^^
KeyError: 860368

**Reply by zdsjjtTLG** [reply]
你的路网点层不是用[gotrackit](https://github.com/zdsjjtTLG/TrackIt)依据线层生产的把？检查一下node层中是否有部分节点没有在线层的from_node和to_node中出现,如果有，请把这部分节点删掉

**Reply by zdsjjtTLG** [reply]
used_node = set(link_gdf['from_node']) | set(link_gdf['to_node'])
    node_gdf.reset_index(inplace=True, drop=True)
    node_gdf.drop(index=node_gdf[~node_gdf['node_id'].isin(used_node)].index, inplace=True, axis=0)
    node_gdf.reset_index(inplace=True, drop=True)

**Reply by Zeitreisender11** [reply]
已解决，感谢！

**Comment by buaa-csc** [comment]
路网优化对数据的坐标系有要求吗

**Reply by zdsjjtTLG** [reply]
输入是EPSG:4326就行

**Reply by ren-max11** [reply]
ID重映射    会输出新的link和node吗？ 为什么没有报错，但是找不到修改后的文件

**Reply by zdsjjtTLG** [reply]
不会返回新的数据，在你传入的link和node上原地修改

**Reply by ren-max11** [reply]
没有报错，但是原文件没有被修改，是因为出现NAN值吗？
    link_id  from_node  to_node
0    221750      22981    82145
1    221751      22981   279426
2     72451      39315    62315
3     45519      39734   258820
4        61      43342   237174
..      ...        ...      ...
81   200904     279426   260740
82    16489     280621   121145
83    57304     287334   181756
84    57303     287334   207853
85   117189     289976    49165

[86 rows x 3 columns]
    node_id
0     12539
1     22981
2     39315
3     39788
4     43342
..      ...
59   261309
60   269563
61   274903
62   279426
63   287334

[64 rows x 1 columns]
    link_id  from_node  to_node
0         1       33.0     44.0
1         2       33.0      2.0
2         3       12.0     51.0
3         4        NaN      3.0
4         5       36.0     58.0
..      ...        ...      ...
81       82        2.0      4.0
82       83        NaN     27.0
83       84       49.0     62.0
84       85       49.0     55.0
85       86        NaN      9.0

[86 rows x 3 columns]
    node_id
0        61
1        33
2        12
3        52
4        36
..      ...
59       29
60       37
61       40
62        2
63       49

[64 rows x 1 columns]

Process finished with exit code 0

**Reply by zdsjjtTLG** [reply]
这不是被修改了吗？？？node_id本来都是5位数，现在不是变成两位数了吗？你的原来的link表的from_node、to_node的值有一部分在node表里面找不到..........才会导致空值

**Reply by ren-max11** [reply]
代码看起来确实修改了，但是打开输入的link和node，还是原来的id

**Reply by zdsjjtTLG** [reply]
你肯定要自己再存一次新文件呀

**Reply by ren-max11** [reply]
哦哦哦，sorry，明白了

**Comment by xueDinge-0502** [comment]
在做路网重塑时，出现了
UnsupportedGEOSVersionError("'remove_repeated_points' requires at least GEOS 3.11.0.")
这个该如何解决呢，谢谢

**Reply by zdsjjtTLG** [reply]
应该是geopandas的版本低了，升级一下geopandas和shapely

**Reply by xueDinge-0502** [reply]
谢谢，升级了版本解决了。

**Comment by Wbc234** [comment]
##########   Don't Enable Region Restrictions，清洗路网线层数据时为啥会显示这个，说我不要启用区域限制，谢谢博主

**Reply by zdsjjtTLG** [reply]
他的意思是：当前没有启用区域限制。不是“不要”的意思

**Reply by Wbc234** [reply]
好的，谢谢!

**Comment by wuhao-ai** [comment]
AttributeError: 'GeoSeries' object has no attribute 'is_closed'
使用clean_link_geo的过程中，出现这个问题的原因是什么呢？

**Reply by zdsjjtTLG** [reply]
应该是geopandas的版本低了，升级一下geopandas和shapely

**Comment by wuhao-ai** [comment]
还是不行哎，都是最新版本的




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年5月3日(星期六) 晚上7:54
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [zdsjjtTLG/TrackIt] en/latest/UserGuide/%E8%B7%AF%E7%BD%91%E4%BC%98%E5%8C%96/ (Discussion #8)














 
应该是geopandas的版本低了，升级一下geopandas和shapely

**Reply by zdsjjtTLG** [reply]
估计你配错环境了

**Comment by wuhao-ai** [comment]
从0开始配置了环境，还是不行哦，改天再试试吧。




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年5月3日(星期六) 晚上8:09
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [zdsjjtTLG/TrackIt] en/latest/UserGuide/%E8%B7%AF%E7%BD%91%E4%BC%98%E5%8C%96/ (Discussion #8)














 
估计你配错环境了

**Reply by zdsjjtTLG** [reply]
你理解错意思了。我不是说你安装的包版本不对，而是你可能在A环境更新了相关包，但是你跑程序用的是B环境，你一直在A环境下更新你的包，这对B环境是不起作用的。**确保你安装更新包使用的环境和你实际跑程序的环境是同一个环境**。

**Comment by wuhao-ai** [comment]
使用clean_link_geo函数进行线层优化时，报错 “ ##########   Don't Enable Region Restrictions ” 是什么原因，没有找到这个错误提示对应的文档。

**Reply by zdsjjtTLG** [reply]
这个不是报错

**Comment by icebear0412** [comment]
拓扑优化一定要做吗
